pkg/snapshot/filesystem.go

```diff
@@ -848,3 +932,137 @@ func GetCurrentFileSystemConfig(ctx context.Context, namespace string) (*types.F
 
 	return &fileSystemConfig, nil
 }
+
+// RevertToMinioFS will apply the spec of the previous BSL to the current one and then update.
+// Used for recovery during a failed migration from Minio to LVP.
+func RevertToMinioFS(ctx context.Context, kotsadmNamespace, veleroNamespace string, previousBsl *velerov1api.BackupStorageLocation) error {
+	bsl, err := FindBackupStoreLocation(context.TODO(), kotsadmNamespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to find backupstoragelocations")
+	}
+
+	bsl.Spec = previousBsl.Spec
+
+	err = UpdateBackupStorageLocation(ctx, veleroNamespace, bsl)
+	if err != nil {
+		return errors.Wrap(err, "failed to revert to minio backup storage location")
+	}
+	return nil
+}
+
+// DeleteFileSystemMinio cleans up the minio resources for hostpath and nfs snapshot deployments.
+// The secret is not deleted, just in case.
+func DeleteFileSystemMinio(ctx context.Context, kotsadmNamespace string) error {
+	clientset, err := k8sutil.GetClientset()
```

Comment:
it's always a good idea to just pass down the original clientset to avoid creating new ones on the fly.